test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,409 @@
+                  prefix_size, characters_type::ALPHABET);
+
+                /* Call search near with the prefix cursor. */
+                cursor_prefix->set_key(cursor_prefix.get(), generated_prefix.c_str());
+                ret = cursor_prefix->search_near(cursor_prefix.get(), &exact_prefix);
+                testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+                if (ret == 0) {
+                    testutil_check(cursor_prefix->get_key(cursor_prefix.get(), &key_prefix));
+                    key_prefix_str = key_prefix;
+                } else {
+                    key_prefix_str = "";
+                }
+
+                /* Open a cursor with the default configuration on the selected collection. */
+                scoped_cursor cursor_default(tc->session.open_scoped_cursor(coll.name.c_str()));
+
+                /* Verify the prefix search_near output using the default cursor. */
+                validate_prefix_search_near(
+                  ret, exact_prefix, key_prefix_str, cursor_default, generated_prefix);
+
+                tc->transaction.add_op();
+                tc->transaction.try_rollback();
+                tc->sleep();
+            }
+            testutil_check(cursor_prefix->reset(cursor_prefix.get()));
+        }
+        /* Roll back the last transaction if still active now the work is finished. */
+        if (tc->transaction.active())
+            tc->transaction.rollback();
+    }
+
+    private:
+    /* Validate prefix search_near call outputs using a cursor without prefix key enabled. */
+    void
+    validate_prefix_search_near(int ret_prefix, int exact_prefix, const std::string &key_prefix,
+      scoped_cursor &cursor_default, const std::string &prefix)
+    {
+        /* Call search near with the default cursor using the given prefix. */
+        int exact_default;
+        cursor_default->set_key(cursor_default.get(), prefix.c_str());
+        int ret_default = cursor_default->search_near(cursor_default.get(), &exact_default);
+
+        /*
+         * It is not possible to have a prefix search near call successful and the default search
+         * near call unsuccessful.
+         */
+        testutil_assert(
+          ret_default == ret_prefix || (ret_default == 0 && ret_prefix == WT_NOTFOUND));
+
+        /* We only have to perform validation when the default search near call is successful. */
+        if (ret_default == 0) {
+            /* Both calls are successful. */
+            if (ret_prefix == 0)
+                validate_successful_calls(
+                  ret_prefix, exact_prefix, key_prefix, cursor_default, exact_default, prefix);
+            /* The prefix search near call failed. */
+            else
+                validate_unsuccessful_prefix_call(cursor_default, prefix, exact_default);
+        }
+    }
+
+    void
+    validate_successful_calls(int ret_prefix, int exact_prefix, const std::string &key_prefix,
```

Comment:
Nitpick: Comment here

Reply:
Added in 4351b568b60309381f7078fdf9761f4534d9ecaa